### ISSUE TITLE
[BUG]: Retry Faliled SubWorkflow,  the Parent Workflow is Completed, but Ancestor Workflow is still Failed!

### ISSUE THREAD

**Issue by xiaowei582648206** [Netflix/conductor#2338]
**Description of  Phenomenon**

I created a three level nested workflow,  a SUB_WORKFLOW task contains another SUB_WORKFLOW task。 
In the thrid level  workflow( **son workflow** ),  the last task  was executed failed ,  I retried the **son workflow**,  it was changed into RUNNING status immediately,  so as the second workflow ( **parent workflow** )。 After some minutes, the **son workflow** and **parent workflow**  executed completed, and changed into COMPLETED status, but the first level workflow ( **ancestor workflow**) is still in FAILED status, and the rest tasks of **ancestor workflow** never schedule.

**Here some screenshots blew**

1.The last task in **son workflow** executed failed 
![image](https://user-images.githubusercontent.com/5030860/124089029-16ec7480-da86-11eb-9512-a8d3a991777a.png)

2.After click **retry** button in **son  workflow** 
![image](https://user-images.githubusercontent.com/5030860/124089329-6337b480-da86-11eb-97be-14fff075a84e.png)

3.The **son and parent workflow** executed completely
![image](https://user-images.githubusercontent.com/5030860/124091462-721f6680-da88-11eb-8135-b3e0f6c7d876.png)

**Questions**
1. Is this a normal phenomenon, or a bug?
2. How can schedule the rest tasks of a**ncestor workflow** by HTTP API, if it is a normal phenomenon?

Thank you very much for your help.

@v1r3n
@apanicker-nflx
@kishorebanala
@jfernandez

**Comment by aravindanr** [comment]
@xiaowei582648206 There are differences in SubWorkflow processing between 2.31.x and 3.0.x, can you provide the version that you are using?

**Comment by xiaowei582648206** [comment]
@aravindanr , thank you very much，We are using version 2.31.0,  next we'll do some test on version 3.0.x,  thank you again!

**Comment by aravindanr** [comment]
> Is this a normal phenomenon, or a bug?

This is a bug, but quite a tricky one to fix. We had to rewrite the SubWorkflow execution logic to fix it in 3.0.x. That's one of the reasons, it can't be easily ported to 2.31.x.

> How can schedule the rest tasks of ancestor workflow by HTTP API, if the is a normal phenomenon?

The ancestor workflow is most likely stuck because it is out of the workflow evaluation cycle. You can re-evaluate the ancestor workflow using the `/sweep/requeue/{workflowId}` api.

**Comment by xiaowei582648206** [comment]
@aravindanr , thank you very much , considering what you said ，we are going to upgrade into v3.0.x, but there is a problem: the version of elasticsearch in product environment is 6.5 , it is old than then 6.8.12 ( the version of ES Client in conductor V3.0.x). what influence will it cause, and which version is recommended for v3.0.x ?

**Comment by aravindanr** [comment]
> what influence will it cause,

Unfortunately, we have only tested with 6.8.x and can't comment on the behavior for 6.5.

> which version is recommended for v3.0.x 

Technically, the `es6-persistence` module is expected to work with **all** `6.x` versions, but as i said we have only tested it with `6.8.x`

fwiw, i think the module should work with 6.5 without any issues. If you run into any, we can support as much as we can. I encourage you to try running Conductor with a docker image of the exact version that your production environment uses.
